CDCgov/phinvads-go
# Add direnv and improve Docker database setup

## Summary

This PR adds `direnv` into the project setup to help us better manage environment variables on a per-project basis. It also makes a few minor changes to the project's `docker-compose.yml` file and adds a `makefile` to streamline the process of starting/stopping the database as well as loading in and refreshing your data using the `phinvads.dump` file.

## Setup

1. Add `DB_PASSWORD=phinvads` to your `.env` file
2. [Install `direnv`](https://direnv.net/docs/installation.html) (`brew install direnv`)
3. [Add hook for your shell](https://direnv.net/docs/hook.html)
4. Restart your terminal
5. Run `direnv allow` in the project directory

## Testing

1. Shut down your local DB if it's running
2. Add `phinvads.dump` into the project directory
3. Run `make start` to start your containerized DB
4. Run `make refresh` to create the DB and load in the data
5. Ensure the app can connect to the DB
6. Run `make stop` to shut down the container when you're done

## Open questions
1. Are we fine with expecting `phinvads.dump` to be in the top-level project directory? Would another location make more sense?
